src/pages/map/map/Map.tsx

```diff
@@ -27,9 +27,19 @@ export default function Map() {
     libraries: ["places"],
   });
 
+  const { style } = useMapStyle();
+
   useEffect(() => {
-    loader.load().then((google) => {
+    const loadMap = async () => { // Define una función asincrónica dentro de useEffect
+      const google = await loader.load(); // Espera a que el Loader cargue la API de Google Maps
+  
       if (mapRef.current) {
+        // Utiliza import() para cargar dinámicamente el archivo de estilos
+        const mapStyles = style === 'dark' 
```

Comment:
can you move this out of the useEffect?

Reply:
The style variable comes from the context, which can change during the life cycle of the component. By placing mapStyles inside the useEffect, we ensure that it is recalculated each time the style changes.